cwatch.php

```diff
@@ -252,57 +182,119 @@ public function getEmailTags()
-            $client = $this->Clients->get($vars['client_id'], false);
-            $firstname = $client->first_name;
-            $lastname = $client->last_name;
-            $email = $client->email;
-            $country = $client->country;
-            $this->loadApi($user, $pass, $sandbox);
-            try{
-                $response = $this->api->createlicence($email, $firstname, $lastname, $country, $product, $term);
-                $json = json_decode($response->resp);
+            $response = null;
+            try {
+                // Log request data
+                $this->log('createcommand', serialize($vars), 'input', true);
+
+                // Add a customer account in cWatch
+                $response = $api->addUser(
+                    $vars['cwatch_email'],
+                    $vars['cwatch_firstname'],
+                    $vars['cwatch_lastname'],
+                    $vars['cwatch_country']
+                );
+
+                // Log response
+                $this->log('adduser', serialize($response), 'output', $response->code == 200);
+
                 if ($response->code != 200) {
-                    $this->Input->setErrors(['api' => ['internal' => $response->errorMsg]]);
+                    $errors = ['api' => ['internal' => $response->errorMsg]];
+                } else {
+                    $license_types = Configure::get('cwatch.products');
+                    // Get a count of licenses to provide for each type
+                    foreach ($license_types as $license_type => $value) {
+                        $license_types[$license_type] = isset($vars['configoptions'][$license_type])
+                            ? $vars['configoptions'][$license_type]
+                            : 0;
+                    }
+
+                    // Determine what term to add these licenses for
+                    $license_term = 'MONTH_1';
+                    if (($pricing = $this->Services->getPackagePricing($vars['pricing_id']))) {
+                        $license_term = strtoupper($pricing->period) . '_' . $pricing->term;
+                    }
+
+                    // Add licenses to the customer account according to the config options provided
+                    foreach ($license_types as $license_type => $quantity) {
+                        for ($i = 0; $i < $quantity; $i++) {
+                            $license_response = $api->addLicense(
+                                $license_type,
+                                $license_type == 'BASIC_DETECTION' ? 'UNLIMITED' : $license_term,
+                                $vars['cwatch_email'],
+                                $vars['cwatch_firstname'],
+                                $vars['cwatch_lastname'],
+                                $vars['cwatch_country']
+                            );
+
+                            $this->log(
+                                'addlicense',
+                                serialize($license_response),
```

Comment:
Same as my last comment on logging the raw data versus the CwatchResponse object. This should be applied to all places we do logging, and any expected sensitive information should be redacted in the log